pkg/build/pipelines/split/bin.yaml

```diff
@@ -0,0 +1,38 @@
+name: Split executable files
+
+needs:
+  packages:
+    - busybox
+
+inputs:
+  package:
+    description: |
+      The package to split executable files from
+    required: false
+
+pipeline:
+  - runs: |
+      pkgname="${{context.name}}"
+      pkgnobin="${pkgname%%-bin}"
+      pkgnobindir="${{targets.outdir}}/$pkgnobin"
+
+      if [ -n "${{inputs.package}}" ]; then
+        PACKAGE_DIR="${{targets.outdir}}/${{inputs.package}}"
+      elif [ "$pkgname" != "$pkgnobin" ] && [ -d "$pkgnobindir" ]; then
+        PACKAGE_DIR="$pkgnobindir"
+      else
+        PACKAGE_DIR="${{targets.destdir}}"
+      fi
+
+      if [ "$PACKAGE_DIR" == "${{targets.contextdir}}" ]; then
+        echo "ERROR: Package can not split files from itself!"
+        exit 1
+      fi
+
+      dir=usr/bin
+      if [ -e "$PACKAGE_DIR/$dir" ] || [ -L "$PACKAGE_DIR/$dir" ]; then
+        d="${{targets.contextdir}}/${i%/*}" # dirname $dir
+        mkdir -p "$d"
+        mv "$PACKAGE_DIR/$dir" "$d"
```

Comment:
would it make more sense to move the contents of usr/bin/* rather than usr/bin itself?  That'd allow you to use this pipeline after something else had already created $d.

If you dont want to do that, then lets fail if $d exists. othereise you'll end up with the `mv` here just moving a 'bin' into usr/bin resulting in `usr/bin/bin/stuff-here`.


if you _do_ want to move the content of usr/bin rather than the directory itself (`mv $PACKAGE_DIR/$dir`)